src/components/VideoQueueItem.vue

```diff
@@ -50,8 +50,23 @@ export default {
 			isLoadingVote: false,
 			hasBeenAdded: false,
 			thumbnailHasError: false,
+			existsInQueue: false,
 		};
-	},
+	}, 
+	created() { 
+        if (this.item.id === this.$store.state.room.currentSource.id) {
+			this.isLoadingAdd = false;
+			this.hasBeenAdded = true;
+			this.existsInQueue = true;
```

Comment:
should return early

Reply:
What do you mean should return early?

Reply:
It should `return` here, because the for loop doesn't need to run